Integrations/integration-CrowdStrikeFalcon.yml

```diff
@@ -0,0 +1,1095 @@
+    }
+    # Note: True life time is actually 30 mins
+    TOKEN_LIFE_TIME = 28
+    INCIDENTS_PER_FETCH = int(demisto.params().get('incidents_per_fetch', 15))
+    # Remove proxy if not set to true in params
+    if not demisto.params().get('proxy'):
+        del os.environ['HTTP_PROXY']
+        del os.environ['HTTPS_PROXY']
+        del os.environ['http_proxy']
+        del os.environ['https_proxy']
+
+    ''' KEY DICTIONARY '''
+
+    DETECTIONS_BASE_KEY_MAP = {
+        'device.hostname': 'System',
+        'device.cid': 'CustomerID',
+        'hostinfo.domain': 'MachineDomain',
+        'detection_id': 'ID',
+        'created_timestamp': 'ProcessStartTime',
+        'max_severity': 'MaxSeverity',
+        'show_in_ui': 'ShowInUi',
+        'status': 'Status'
+    }
+
+    DETECTIONS_BEHAVIORS_KEY_MAP = {
+        'filename': 'FileName',
+        'scenario': 'Scenario',
+        'md5': 'MD5',
+        'sha256': 'SHA256',
+        'ioc_type': 'IOCType',
+        'ioc_value': 'IOCValue',
+        'cmdline': 'CommandLine',
+        'user_name': 'UserName',
+        'behavior_id': 'ID',
+    }
+
+    SEARCH_IOC_KEY_MAP = {
+        'type': 'Type',
+        'value': 'Value',
+        'policy': 'Policy',
+        'source': 'Source',
+        'share_level': 'ShareLevel',
+        'expiration_timestamp': 'Expiration',
+        'description': 'Description',
+        'created_timestamp': 'CreatedTime',
+        'created_by': 'CreatedBy',
+        'modified_timestamp': 'ModifiedTime',
+        'modified_by': 'ModifiedBy'
+    }
+
+    SEARCH_DEVICE_KEY_MAP = {
+        'device_id': 'ID',
+        'external_ip': 'ExternalIP',
+        'local_ip': 'LocalIP',
+        'hostname': 'Hostname',
+        'os_version': 'OS',
+        'mac_address': 'MacAddress',
+        'first_seen': 'FirstSeen',
+        'last_seen': 'LastSeen'
+    }  # TODO: Add PolicyType via a function - key is policies
+
+    ''' SPLIT KEY DICTIONARY '''
+
+    """
+        Pattern:
+        {
+            'Path': 'Path to item',
+            'NewKey': 'Value of output key',
+            'Delim': 'Delimiter char',
+            'Index': Split Array Index
+        }
+    """
+    DETECTIONS_BEHAVIORS_SPLIT_KEY_MAP = [
+        {
+            'Path': 'parent_details.parent_process_graph_id',
+            'NewKey': 'SensorID',
+            'Delim': ':',
+            'Index': 1
+        },
+        {
+            'Path': 'parent_details.parent_process_graph_id',
+            'NewKey': 'ParentProcessID',
+            'Delim': ':',
+            'Index': 2
+        },
+        {
+            'Path': 'triggering_process_graph_id',
+            'NewKey': 'ProcessID',
+            'Delim': ':',
+            'Index': 2
+        },
+    ]
+
+    ''' HELPER FUNCTIONS '''
+
+
+    def http_request(method, url_suffix, params=None, data=None, headers=HEADERS, safe=False):
+        """
+            A wrapper for requests lib to send our requests and handle requests and responses better.
```

Comment:
@DeanArbel did we start to use sphinx doc strings or was it your initiative?

Reply:
This method specifically is part of the default integration provided by the plugin, but it was my initiative to add docstring (reST format, as this is what is used in commonServerPython and what is generated in the IDE)

Reply:
OK